pkg/skaffold/event/v2/event.go

```diff
@@ -241,16 +241,16 @@ func emptyStatusCheckState() *proto.StatusCheckState {
 	}
 }
 
-func AutoTriggerDiff(name string, val bool) (bool, error) {
-	switch name {
-	case "build":
+func AutoTriggerDiff(phase sErrors.Phase, val bool) (bool, error) {
+	switch phase {
+	case sErrors.Build:
```

Comment:
seems a bit counterintuitive that the the event `phases` are imported from the error package.

Reply:
I agree and have been wondering where a better place to house these would be. Since phases are a broader concept of skaffold, maybe they should be in something more general?

Reply:
would it be better to move the `Phase` definition into the `proto` package?

Reply:
I think I'm not sure on that just because they aren't specific to anything proto related. I would say it almost seems like these should just be consts in the `constants` package? 

@tejal29 any reason for if these should stay in the `errors` package?

Either way, I think this could be handled in another follow up refactor if you're okay with that

Reply:
`Phase` could be an enum in `enums.proto`